library/psa_crypto.c

```diff
@@ -6010,38 +6010,73 @@ static psa_status_t psa_read_rsa_exponent( const uint8_t *domain_parameters,
 }
 #endif /* MBEDTLS_RSA_C && MBEDTLS_GENPRIME */
 
-static psa_status_t psa_generate_key_internal(
-    psa_key_slot_t *slot, size_t bits,
-    const uint8_t *domain_parameters, size_t domain_parameters_size )
+static psa_status_t psa_get_key_size(
```

Comment:
This comment is still relevant to the last version. I addressed the documentation part: see [commit](https://github.com/ARMmbed/mbedtls/pull/3744/commits/f43e1d05a0b9ac83590b34fb08212ca7c7bfbd06). I didn't change the code to use PSA_EXPORT_KEY_OUTPUT_SIZE (previously PSA_KEY_EXPORT_MAX_SIZE) as this would result in a change of buffer size, for example for ECC keys: PSA_KEY_EXPORT_ECC_[KEY_PAIR|PUBLIC_KEY]_MAX_SIZE(key_bits) instead of PSA_BITS_TO_BYTES(key_bits) ... Not sure if I need to address this in this PR, let me know if you think so.

Reply:
I don't understand when there would be a difference in size. For ECC key pairs, it's the bit-size rounded up to a whole number of bytes in both cases.

Reply:
Ah yes sorry, I didn't check the definition of PSA_KEY_EXPORT_ECC_KEY_PAIR_MAX_SIZE(key_bits).

Reply:
I split the additional validation needed for the key type and length for key generation and the key buffer size calculation. The key buffer size calculation is done just using  PSA_EXPORT_KEY_OUTPUT_SIZE now, see [1120cc0](https://github.com/ARMmbed/mbedtls/pull/3744/commits/1120cc08c396bf4c86ae96e73db86c8c25220936).